### ISSUE TITLE
podman run --cgroup-parent: validation

### ISSUE THREAD

**Issue by edsantiago** [projectatomic/libpod#497]
podman does not seem to be validating the argument to `--cgroup-parent`. Should it?

```
# podman run --rm --cgroup-parent=/foo fedora cat /proc/1/cgroup
11:pids:/foo/libpod-conmon-0d9[...]/0d9[...]
10:freezer:/foo/libpod-conmon/0d9[...]/0d9[...]
[...]
1:name=systemd:/foo/libpod-conmon-0d9[...]/0d9[...]
0::/user.slice/user-0.slice/session-1.scope
```

docker prevents this: it explicitly checks (daemon_unix.go) that the argument end in `.slice`:

    # docker run  --cgroup-parent=/foo fedora true
    /usr/bin/docker-current: Error response from daemon: cgroup-parent for systemd cgroup should be a valid slice named as "xxx.slice".

docker (or someone else: oci-systemd-hook? runc? I can't find it) also blocks a leading slash:

    # docker run  --cgroup-parent=/foo.slice fedora true
    container_linux.go:247: starting container process caused "process_linux.go:258: applying cgroup configuration for process caused \"Invalid slice name /foo.slice\""
    /usr/bin/docker-current: Error response from daemon: invalid header field value "oci runtime error: container_linux.go:247: starting container process caused \"process_linux.go:258: applying cgroup configuration for process caused \\\"Invalid slice name /foo.slice\\\"\"\n".

As always with cgroups, I have no idea if these are important.

**Comment by mheon** [comment]
I'll check to see if this is necessary; if it is, I think our defaults may also be bad

**Comment by rhatdan** [comment]
We should validate.

**Comment by mheon** [comment]
We don't use the systemd cgroup manager by default, so our CGroups should explicitly not have .slice in them. I will add validation based on what CRI-O uses. If this is not sufficient, we can add further validation later.

**Comment by mheon** [comment]
Fixed by #507
